elasticdl/python/master/task_dispatcher.py

```diff
@@ -134,6 +142,41 @@ def create_tasks(self, task_type, model_version=-1):
                 self._todo.extend(tasks)
         return tasks
 
+    def _create_save_model_task(self):
+        # For the SavedModel Task, we need a shard of data to build the model
+        shards = self._training_shards
+        assert shards is not None
+
+        (shard_name, (start_ind_this_shard, num_records_this_shard)) = next(
+            iter(shards.items())
+        )
+        start_ind_this_task = start_ind_this_shard
+        end_ind_this_task = start_ind_this_shard + min(
+            self._records_per_task, num_records_this_shard
+        )
+
+        # Use the first shard of data to do the SavedModel work
+        save_model_task = _Task(
+            shard_name=shard_name,
+            start=start_ind_this_task,
+            end=end_ind_this_task,
+            type=elasticdl_pb2.SAVE_MODEL,
+        )
+
+        self._todo.append(save_model_task)
+
+    def invoke_task_list_done_callback(self):
```

Comment:
A lock is needed here, as there may be multiple concurrent GetTask rpc calls.

Reply:
Good catch, fixed :)